kotlinpoet/src/commonTest/kotlin/com/squareup/kotlinpoet/MemberNameTest.kt

```diff
@@ -322,6 +322,87 @@ class MemberNameTest {
     )
   }
 
+  @Test fun importedMemberClassFunctionNameDontClashForParameterValue() {
+    fun createSimpleParameterCodeBlock(): CodeBlock {
+      val member = ClassName(
+        packageName = "com.squareup",
+        simpleNames = listOf("Fridge"),
+      ).member("meat")
+      val block = CodeBlock.builder()
+        .add("%M", member)
+        .add(" { }")
+        .build()
+      return CodeBlock
+        .builder()
+        .add("%L", block)
+        .build()
+    }
+
+    fun createFuncParameter(name: String) =
+      CodeBlock
+        .builder()
+        .add(name).add(" = ")
+        .add(createSimpleParameterCodeBlock())
+        .build()
+
+    fun createFuncParameters(names: List<String>): List<CodeBlock> =
+      names.map { name -> createFuncParameter(name) }
+
+    fun createBuildFunc(params: List<String>): FunSpec {
+      val tacoClassname = ClassName.bestGuess("com.squareup.tacos.Taco")
+      val paramsCodeBlock = CodeBlock.builder()
+        .add("return %T", tacoClassname)
+        .apply {
+          add(
+            params.let(::createFuncParameters)
+              .map { block -> CodeBlock.of("%L", block) }
+              .joinToCode(prefix = "(", suffix = ")"),
+          )
+        }
+        .build()
+      return FunSpec.builder("build")
+        .returns(tacoClassname)
+        .addCode(paramsCodeBlock).build()
+    }
+
+    val spec = FileSpec.builder("com.squareup.tacos", "Tacos")
+      .addType(
+        TypeSpec.classBuilder("DeliciousTaco")
+          .addFunction(createBuildFunc(listOf("deliciousMeat")))
```

Comment:
nit: I'd suggest inlining the helper functions that are only called once, especially since they're more or less one-liners - IMO that would make the code easier to read.